src/ansys/pyensight/core/utils/dsg_server.py

```diff
@@ -55,6 +68,63 @@ def reset(self, cmd: Any = None) -> None:
         if cmd is not None:
             self.hash.update(cmd.hash.encode("utf-8"))
         self.cmd = cmd
+        self._material = None
+
+    def _parse_material(self) -> None:
+        """
+        Parse the JSON string in the part command material string and
+        make the content accessible via material_names() and material().
+        """
+        if self._material is not None:
+            return
+        try:
+            if self.cmd.material_name:  # type: ignore
+                self._material = json.loads(self.cmd.material_name)  # type: ignore
+                for key, value in self._material.items():
+                    value["name"] = key

```

Comment:
What does this do?

Reply:
I think I see.  You know the value is a dict, and it was useful to put the name into the dict.